src/Compilers/CSharp/Test/Semantic/Semantics/InterceptorsTests.cs

```diff
@@ -0,0 +1,1620 @@
+    }
+
+    [Fact]
+    public void InterceptorCannotBeGeneric_03()
+    {
+        var source = """
+            using System.Runtime.CompilerServices;
+            using System;
+
+            interface I1 { }
+            class C : I1
+            {
+                [Interceptable]
+                public static void InterceptableMethod(string param) { Console.Write("interceptable " + param); }
+            }
+
+            static class Program
+            {
+                public static void Main()
+                {
+                    C.InterceptableMethod("call site");
+                }
+            }
+
+            static class Outer<T>
+            {
+                static class D
+                {
+                    [InterceptsLocation("Program.cs", 15, 11)]
+                    public static void Interceptor1(string param) { Console.Write("interceptor " + param); }
+                }
+            }
+            """;
+        var comp = CreateCompilation(new[] { (source, "Program.cs"), s_attributesSource });
+        comp.VerifyEmitDiagnostics(
+            // Program.cs(23,10): error CS27001: Method 'Outer<T>.D.Interceptor1(string)' cannot be used as an interceptor because it or its containing type has type parameters.
+            //         [InterceptsLocation("Program.cs", 15, 11)]
+            Diagnostic(ErrorCode.ERR_InterceptorCannotBeGeneric, @"InterceptsLocation(""Program.cs"", 15, 11)").WithArguments("Outer<T>.D.Interceptor1(string)").WithLocation(23, 10)
+            );
+    }
+
+    [Fact]
+    public void InterceptableGeneric_01()
+    {
+        var source = """
+            using System.Runtime.CompilerServices;
+            using System;
+
+            class C
+            {
+                [Interceptable]
+                public static void InterceptableMethod<T>(T t) { Console.Write("0"); }
+            }
+
+            static class Program
+            {
+                public static void Main()
+                {
+                    C.InterceptableMethod<string>("1");
+                    C.InterceptableMethod("2");
+                }
+            }
+
+            static class D
+            {
+                [InterceptsLocation("Program.cs", 14, 11)]
+                [InterceptsLocation("Program.cs", 15, 11)]
+                public static void Interceptor1(string s) { Console.Write(s); }
+            }
+            """;
+        var verifier = CompileAndVerify(new[] { (source, "Program.cs"), s_attributesSource }, expectedOutput: "12");
+        verifier.VerifyDiagnostics();
+    }
+
+    [Fact]
+    public void InterceptableGeneric_02()
+    {
+        var source = """
+            using System.Runtime.CompilerServices;
+            using System;
+
+            class C
+            {
+                [Interceptable]
+                public static void InterceptableMethod<T>(T t) { Console.Write("0"); }
```

Comment:
>Console.Write("0"); [](http://example.com/codeflow?start=65&length=19)

nit: for test codepaths that are unreachable, consider throwing instead (in the future).